lib/contracts/genesisProtocol.js

```diff
@@ -0,0 +1,671 @@
+    }
+
+    const avatar = await this.contract.proposalAvatar(
+      options.proposalId
+    );
+
+    return avatar;
+  }
+
+  async getScoreThresholdParams(opts = {}) {
+
+    const defaults = {
+      avatar: undefined,
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.avatar) {
+      throw new Error("avatar is not defined");
+    }
+
+    const result = await this.contract.scoreThresholdParams(
+      options.avatar
+    );
+
+    // TODO:  convert to number??  dunno what these values represent.
+    return {
+      thresholdConstA: result[0],
+      thresholdConstB: result[1]
+    };
+  }
+
+  async getStakerInfo(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined,
+      staker: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    if (!options.staker) {
+      throw new Error("staker is not defined");
+    }
+
+    const result = await this.contract.staker(
+      options.proposalId
+      , options.staker
+    );
+
+    return {
+      vote: result[0].toNumber(),
+      stake: result[1]
+    };
+  }
+
+  async getVoteStake(opts = {}) {
+    const defaults = {
+      proposalId: undefined,
+      vote: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    this._validateVote(options.vote, options.proposalId);
+
+    const stake = await this.contract.voteStake(
+      options.proposalId
+      , options.vote
+    );
+
+    return stake;
+  }
+
+  async getWinningVote(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    const winningVote = await this.contract.winningVote(
+      options.proposalId
+    );
+
+    return winningVote.toNumber();
+  }
+
+  async getState(opts = {}) {
+
+    const defaults = {
+      proposalId: undefined
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.proposalId) {
+      throw new Error("proposalId is not defined");
+    }
+
+    const state = await this.contract.state(
+      options.proposalId
+    );
+
+    return state;
+  }
+
+  async _validateVote(vote, proposalId) {
+    const numChoices = await this.getNumberOfChoices({ proposalId: proposalId });
+    if (!Number.isInteger(vote) || (vote < 0) || (vote >= numChoices)) {
+      throw new Error("vote is not valid");
+    }
+  }
+
+  /**
+   * Set the contract parameters.
+   * @param {GenesisProtocolParams} params
+   * @returns parameters hash
+   */
+  async setParams(params) {
+
+    params = Object.assign({},
+      {
+        preBoostedVoteRequiredPercentage: 50,
```

Comment:
where do these defaults come from?

Reply:
Good question.  I pulled them from some Arc test code.  I would like to know what they should be  @orenyodfat

Reply:
I am not sure there is a solid decision re default values yet.
any how ,it can be override by the caller party.

Reply:
I'll leave them as is until a decision is made to change them